cos/static/board_builder.js

```diff
@@ -130,11 +142,100 @@ function build_board() {
 
                 layer.add(settlement_area_right);
             }
+            settlement_area_right.on('mouseup', function(){
+                if(settlement_area_right.getFill() == 'red'){
+                    place_settlement(this.x(),this.y(), settlementX, settlementY, stage, layer);
+                }
+            })
+
+        }
+    }
 
+    for (var i = 0; i < 6; i++) {
+        var settlement = new Konva.Shape({
+                x: settlementX,
+                y: settlementY,
+           sceneFunc: function (context) {
+               context.beginPath();
+               context.moveTo(-7, 4);
+               context.lineTo(-7, -10);
+               context.lineTo(0, -17);
+               context.lineTo(7, -10);
+               context.lineTo(7, 4);
+               context.lineTo(-7, 4);
+               context.closePath();
+
+               context.fillStrokeShape(this);
+               },
+               fill: 'red',
+               stroke: 'black',
+               strokeWidth: 1,
+               name : 'settlement'
+        });
+        if (i==5) {
+            settlement.on('mousedown', function(){
+                mark_settlement_placement(stage,layer,false, settlementX, settlementY);
+            })
+            settlement.id('settlement_button');
         }
+        layer.add(settlement);
     }
 
+    get_player_info(update_settlement_color,stage, layer);
+
 
     // add the layer to the stage
     stage.add(layer);
+}
+
+
+//Redraw settlements with info from backend
+function update_settlement_color(data,stage,layer) {
+    var players = data.Players;
+        var settlements = stage.find('.settlement');
+        for (i = 0; i < 6; i++){
+            settlements[i].fill(players[0].Player.player_color);
+            console.log("fughetaboutit");
+            layer.batchDraw();
+        }
+}
+
+//Illuminates legal settlement locations for placement
```

Comment:
Can you add a quick definition of what a legal settlement location is? If I am reading the code correctly, it looks like anyplace that is without an existing settlement but I am not 100% sure.